azurelinuxagent/ga/exthandlers.py

```diff
@@ -768,6 +768,42 @@ def report_ext_handler_status(self, vm_status, ext_handler):
 
 
 class ExtHandlerInstance(object):
+
+    def __truncate_file_head(self, filename, max_size):
+        try:
+            if os.stat(filename).st_size <= max_size:
+                return
+            
+            with open(filename, "rb") as existing_file:
+                existing_file.seek(-1 * max_size, 2)
+                _ = existing_file.readline()
+
+                with open(filename + ".tmp", "wb") as tmp_file:
+                    shutil.copyfileobj(existing_file, tmp_file)
+
+            os.rename(filename + ".tmp", filename)
+
+        except (IOError, OSError) as e:
+            if is_file_not_found_error(e):
+                # If CommandExecution.log does not exist, it's not noteworthy;
+                # this just means that no extension with self.ext_handler.name is
+                # installed.
+                return
+            
+            logger.error("Exception occurred while attempting to truncate CommandExecution.log "
+                "for extension {0}. Exception is: {1}", self.ext_handler.name, e)
+
+            for f in (filename, filename + ".tmp"):
```

Comment:
Small NIT: Might be worthwhile to make this into a small utility function in fileutil.py so that we can use it in other places too 

Something like - 

> fileutil.rmfile(ignore-errors=True)

I'm good either way though